docs/docs/multirole.md

```diff
@@ -0,0 +1,337 @@
+
+```
+echo '{"role": "dev"} | curl -X PATCH http://local:8080/v2/apps/dev/bigbusiness -H "Content-Type: application/json" --data "@-"
+```
+
+This will trigger a deployment where all of the old bigbusiness instances running as the role `"slave_public"` are killed, and replaced by new instances running as the role "dev". Further, Marathon will update its FrameworkInfo information with Mesos to also include the role "dev", so that it can begin receiving offers allocated to the role. This may require the Mesos create role permission if the role does not already exist. See the section below on Mesos permission caveats.
+
+## Resident service multi-role migration
+
+Migrating resident services is a bit different, since changing the role for reserved resources would result in the reservations being destroyed (and subsequently the volume data), and this is often not desired. Instead of wiping old reserved instances, Marathon allows a resident service role to be changed for newly created instances, continuing to leave the existing instances running with the old role.
+
+For example, say that we have deployed the following resident service:
+
+```
+$ cat <<EOF | curl -v -X POST http://localhost:8080/v2/pods -H "Content-Type: application/json" --data "@-"
+{
+  "id": "/dev/bigbusiness",
+  "containers": [
+    {
+      "name": "sleep1",
+      "exec": { "command": { "shell": "pwd; ls -la; date >> data/launches.txt; sleep 1000" } },
+      "resources": { "cpus": 0.1, "mem": 32 },
+      "volumeMounts": [{
+        "name": "data",
+        "mountPath": "data"
+      }]
+    }
+  ],
+  "volumes": [
+    {
+      "name": "data",
+      "persistent": {"size": 16, "type": "root"}
+    }
+  ]
+}
+EOF
+```
+
+When we check the roles for the running instances, we will see that the instances were launched with the role `"slave_public":`
+
+```
+$ curl http://localhost:8080/v2/pods/dev/bigbusiness::status | jq '.instances[].role'
+```
+
+When we attempt to PUT the update with the role "dev" to the pods endpoint, as follows:
+
+```
+$ cat <<EOF | curl -v -X PUT http://localhost:8080/v2/pods/dev/bigbusiness -H "Content-Type: application/json" --data "@-"
+{
+  "id": "/dev/bigbusiness",
+  "containers": [
+    {
+      "name": "sleep1",
+      "exec": { "command": { "shell": "pwd; ls -la; date >> le-data/launches.txt; sleep 1000" } },
+      "resources": { "cpus": 0.1, "mem": 32 },
+      "volumeMounts": [{
+        "name": "data",
+        "mountPath": "le-data"
+      }]
+    }
+  ],
+  "volumes": [
+    {
+      "name": "data",
+      "persistent": {"size": 16, "type": "root"}
+    }
+  ],
+  "role": "dev"
+}
+EOF
+```
+
+... the API request will fail, indicating that resident services cannot change the role for existing instances. In our case, we would like to proceed, so that new instances are allocated to the new role, so we can try again with the parameter "? force=true":
```

Comment:
```suggestion
... the API request will fail, indicating that resident services cannot change the role for existing instances. In case you would like to proceed, so that new instances are allocated to the new role, then you can do so by appending  the parameter "?force=true":
```